model-optimizer/extensions/middle/split_tdnn_memoryoffset.py

```diff
@@ -0,0 +1,46 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from mo.graph.graph import Graph
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.memoryoffset import MemoryOffset
+from mo.ops.result import Result
+
+
+class SplitTdnnMemoryOffset(MiddleReplacementPattern):
+    '''
+    Splits MemoryOffsets in TDNN blocks into 2 parts. These parts then will be converted to ReadValue and Assign.
+    '''
+    enabled = True
+    run_not_recursively = True
+
+    def run_after(self):
+        from extensions.middle.PartialInfer import PartialInfer
+        return [PartialInfer]
+
+    def run_before(self):
+        from extensions.middle.ReplaceMemoryOffsetWithSplice import ReplaceMemoryOffsetWithMemoryNodePattern, ReplaceMemoryOffsetNodePattern
+        return [ReplaceMemoryOffsetNodePattern, ReplaceMemoryOffsetWithMemoryNodePattern]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for offset_node in graph.get_op_nodes(op='MemoryOffset', splitted=False):
+            paired_node = MemoryOffset(graph, {'name': offset_node.pair_name, 'splitted': True, 'pair_name': offset_node.id,
+                                               't': offset_node.t, 'has_default': offset_node.has_default}).create_node()
+            offset_node['splitted'] = True
+            offset_node.out_port(0).get_connection().set_source(paired_node.out_port(0))
+            res_node = Result(graph, {'name': offset_node.id + "_output"}).create_node()
+            offset_node.out_port(0).connect(res_node.in_port(0))
+            paired_node['need_shape_inference'] = False
```

Comment:
Why do you disable shape inference?

Reply:
If we do not set this attribute to `False` then after transformation it will try to `PartialInfer` and will fail because splitted node does not have `element_size` attribute. We either must (I) declare it here or (II) disable `PartialInfer` here. The first option seems redundant because data nodes on out_port 0 already have adequate shapes so I disabled `PartialInfer` for this node.

Reply:
But what if sometime in the future someone will call shape inference for this node? It will fail and MO will crash. So let's set all necessary attributes so the shape infer will not fail.

Reply:
Added setting of `element_size` here. Now if someone will call shape infer for this node it will not fail.